### PR TITLE
Adding CircleCI Job to trigger test_upgrade on RC build

### PR DESCRIPTION
**Pull request by mishah334**
## Description

Adding CircleCI Job extending workflow `build-and-release-helm-chart` to trigger `test_upgrade` workflow of `terraform-aws-astronomer` on RC build.

## Related Issues

Issue: https://github.com/astronomer/issues/issues/5461

## Testing

As this is the CircleCI Workflow, it can be tested as execution.

## Master PR

https://github.com/astronomer/astronomer/pull/1835

### CODE REVIEW

File: bin/trigger_rc_tests.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if "success" != pipeline_state and "running" != pipeline_state:
+        raise SystemError(1)
+    else:
+        raise SystemExit(0)
+
+
+if __name__ == "__main__":
+    arg_parser = argparse.ArgumentParser(description="Optional app description")
+
+    # Required positional argument
```
Comment (danielhoherd) [inline bin/trigger_rc_tests.py:119]: Remove unneeded comment. `required=True` is sufficient information.

```suggestion
    # Required positional argument
```

File: bin/trigger_rc_tests.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    # Run Workflow
+    run_workflow_resp = run_workflow(
+        circleci_token=circleci_token, parameters=parameters
+    )
+
+    pipeline_id = json.loads(run_workflow_resp)["id"]
+    pipeline_number = json.loads(run_workflow_resp)["number"]
+
+    # Printing Info
```
Comment (danielhoherd) [inline bin/trigger_rc_tests.py:91]: Remove unneeded comment. The function call is nearly the same as this comment.
```suggestion
```

File: bin/trigger_rc_tests.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            astro_version = file_name
+
+    if astro_version is None:
+        print(
+            f"INFO: Skipping calling workflow as no valid version found for the RC test. Below files are found at path: {astro_path}."
+        )
+        print(json.dumps(file_list))
+        raise SystemExit(0)
+
+    # Trigger the workflow
```
Comment (danielhoherd) [inline bin/trigger_rc_tests.py:73]: Remove ambiguous comment. It talks about triggering a workflow, but the next two lines are string manipulation, then dict creation, then information printing.
```suggestion
```

File: bin/trigger_rc_tests.py (hunk 1 above)
Comment (mishah334) [reply bin/trigger_rc_tests.py]: Done!

**Pull request merged**